pkg/helpers/helpers.go

```diff
@@ -160,3 +161,21 @@ func ConcatSlices[T any](slices ...[]T) []T {
 	}
 	return tmp
 }
+
+// TODO(SNOW-1058419): address during identifiers rework
+func ParseRootLocation(location string) (sdk.SchemaObjectIdentifier, string, error) {
+	location = strings.TrimPrefix(location, "@")
+	parts, err := parseIdentifierStringWithOpts(location, func(r *csv.Reader) {
+		r.Comma = '/'
+		r.LazyQuotes = true
+	})
+	if err != nil {
+		return sdk.SchemaObjectIdentifier{}, "", err
+	}
+	if len(parts) < 3 {
+		return sdk.SchemaObjectIdentifier{}, "", fmt.Errorf("expected 3 parts, got ")
+	}
+	parts[2] = strings.Join(parts[2:], ".")
+	lastParts := strings.Split(parts[2], "/")
+	return sdk.NewSchemaObjectIdentifier(parts[0], parts[1], lastParts[0]), strings.Join(lastParts[1:], "/"), nil
```

Comment:
I kind of don't get this part fully (please add tests)

Reply:
Added.